src/components/TodoList.jsx

```diff
@@ -13,51 +13,121 @@ class TodoListXP extends React.Component {
 
         this.state = {
             tasks: [
-                "do one thing",
-                "another one",
-                "last one to do"
+                { taskName: "do one thing", date: new Date() },
+                { taskName: "another one", date: new Date() },
+                { taskName: "last one to do", date: new Date() }
             ],
-            input: "nichego net"
+            input: ""
         };
-
         this.addTask = this.addTask.bind(this);
         this.removeTask = this.removeTask.bind(this);
         this.onInputChange = this.onInputChange.bind(this);
+        this.removeAllTasks = this.removeAllTasks.bind(this)
+        this.sortTasks = this.sortTasks.bind(this)
     }
 
-    removeTask(text) {
-        const { tasks } = this.state;
 
+    sortTasks(event) {
+        const { tasks } = this.state
+        const option = event.target.value
+        let newState
+        if (option === "Name asc") {
+            newState = {
+                ...this.state,
+                tasks: tasks.sort((task1, task2) => (task1.taskName.toLowerCase() > task2.taskName.toLowerCase()) ? 1 : -1)
+            }
+        }
+        if (option === "Name desc") {
+            newState = {
+                ...this.state,
+                tasks: (tasks.sort((task1, task2) => (task1.taskName.toLowerCase() > task2.taskName.toLowerCase()) ? 1 : -1)).reverse()
+            }
+        }
+        if (option === "Newer") {
+            newState = {
+                ...this.state,
+                tasks: (tasks.sort((task1, task2) => (task1.date - task2.date))).reverse()
+            }
+        }
+        if (option === "Older") {
+            newState = {
+                ...this.state,
+                tasks: tasks.sort((task1, task2) => (task1.date - task2.date))
+            }
+        }
 
-        const newState = {
+        this.setState(newState)
+    }
+
+
+    taskTexts = () => {
+        const taskNames = []
+        for (const task of this.state.tasks) {
+            taskNames.push(task.taskName)
+        }
+        return taskNames
+    }
```

Comment:
When you are writing `class methods`, please stick to other syntax (not lambdas). This one is slower, cause with each new class instantiation it creates all these methods again (IIRC)
So 

```js
taskTexts() {
  ...
}
```